server/controllers/finance/reports/cashflow/index.js

```diff
@@ -99,51 +103,33 @@ function queryIncomeExpense(params, dateFrom, dateTo) {
   }
 
   const requette = `
-      SELECT BUID(t.uuid) AS uuid, t.trans_id, t.trans_date, a.number, a.label,
-        SUM(t.debit_equiv) AS debit_equiv, SUM(t.credit_equiv) AS credit_equiv,
-        t.debit, t.credit, t.currency_id, t.description, t.comment,
-        BUID(t.record_uuid) AS record_uuid, t.origin_id, u.display_name,
-        x.text AS transactionType
-      FROM
-      (
-        (
-          SELECT pj.project_id, pj.uuid, pj.record_uuid, pj.trans_date,
-            pj.debit_equiv, pj.credit_equiv, pj.debit, pj.credit,
-            pj.account_id, pj.entity_uuid, pj.currency_id, pj.trans_id,
-            pj.description, pj.comment, pj.origin_id, pj.user_id
-          FROM posting_journal pj
-          WHERE pj.account_id IN (?) AND 
-            DATE(pj.trans_date) >= DATE(?) AND 
-            DATE(pj.trans_date) <= DATE(?) AND 
-            pj.origin_id <> 10 AND 
-            pj.record_uuid NOT IN 
-            (SELECT reference_uuid FROM voucher WHERE type_id = 10)
-        ) UNION (
-          SELECT gl.project_id, gl.uuid, gl.record_uuid, gl.trans_date,
-            gl.debit_equiv, gl.credit_equiv, gl.debit, gl.credit,
-            gl.account_id, gl.entity_uuid, gl.currency_id, gl.trans_id,
-            gl.description, gl.comment, gl.origin_id, gl.user_id
-          FROM general_ledger gl
-          WHERE gl.account_id IN (?) AND
-           DATE(gl.trans_date) >= DATE(?) AND
-            DATE(gl.trans_date) <= DATE(?) AND
-             gl.origin_id <> 10 AND
-             gl.record_uuid NOT IN
-             (SELECT reference_uuid FROM voucher WHERE type_id = 10)
-        )
-      ) AS t, account AS a, user as u, transaction_type as x
-      WHERE t.account_id = a.id AND t.user_id = u.id AND t.origin_id = x.id
-      GROUP BY t.trans_id ;`;
+    SELECT BUID(t.uuid) AS uuid, t.trans_id, t.trans_date, t.period_id, a.number, a.label,
+      SUM(t.debit_equiv) AS debit_equiv, SUM(t.credit_equiv) AS credit_equiv,
+      t.debit, t.credit, t.currency_id, t.description, t.comment,
+      BUID(t.record_uuid) AS record_uuid, t.origin_id, u.display_name,
+      x.text AS transactionType
+    FROM (        
+      SELECT gl.project_id, gl.uuid, gl.record_uuid, gl.trans_date, gl.period_id,
+        gl.debit_equiv, gl.credit_equiv, gl.debit, gl.credit,
+        gl.account_id, gl.entity_uuid, gl.currency_id, gl.trans_id,
+        gl.description, gl.comment, gl.origin_id, gl.user_id
+      FROM general_ledger gl
+      WHERE gl.account_id IN (?) AND
```

Comment:
It would be more efficient to say `account_id = ?`.  It doesn't look like you use multiple accounts.